webapp/src/modules/publication/selectors.js

```diff
@@ -1,4 +1,25 @@
+import { createSelector } from 'reselect'
+import { PUBLISH_SUCCESS } from './actions'
+import { getTransactionsByType } from 'modules/transaction/selectors'
+
 export const getState = state => state.publication
-export const getPublications = state => getState(state).data
+export const getData = state => getState(state).data
 export const isLoading = state => getState(state).loading.length > 0
 export const getError = state => getState(state).error
+
+export const getPublications = createSelector(
+  getData,
+  state => getTransactionsByType(state, PUBLISH_SUCCESS),
+  (publications = {}, publishTransactions) => {
+    for (const transaction of publishTransactions) {
+      const publication = transaction.payload
+      publications[publication.tx_hash] = {
```

Comment:
Watch out, this is mutating the app state. The app state should be immutable and new states should only be generated from reducers. We should create a new object here and assign the enriched publications to it, then return that new object.